cms/djangoapps/contentstore/qti_converter.py

```diff
@@ -0,0 +1,385 @@
+
+    seq_f.write(
+        '<sequential display_name="Subsection" due="{0}" start="{1}" show_correctness="{2}" '.
+        format(chapter.end, chapter.start, show_correct))
+    if chapter.timeLimit:
+        seq_f.write('is_time_limited="true" is_proctored_enabled="false" '
+                    'is_practice_exam="false" default_time_limit_minutes="{0}"'.
+                    format(chapter.timeLimit))
+    seq_f.write('>')
+
+    vertical_id = uuid.uuid4().hex
+    vertical_f = open('{0}/{1}.xml'.format(vertical_dir, vertical_id), "w+")
+    vertical_f.write('<vertical display_name="{0}">'.format(chapter.title))
+    if chapter.description:
+        vertical_f.write('<html url_name="{0}"/>'.format(html_id))
+    for item in chapter.items:
+        write_problem(item, vertical_f, chapter, problem_dir)
+
+    vertical_f.write('</vertical>')
+    vertical_f.close()
+    seq_f.write('<vertical url_name="{0}"/>'.format(vertical_id))
+    seq_f.write('</sequential>')
+    seq_f.close()
+    chapter_f.write('</chapter>')
+    chapter_f.close()
+
+
+def write_olx(course_directory, course):
+    """
+    Write parsed OLX file data to file system.
+    """
+    course_f = open('{0}/course.xml'.format(course_directory), 'w+')
+    course_f.write("<course url_name=\"course\" org=\"credo\" course=\"cs1\"/>")
+    course_f.close()
+
+    chapter_dir = '{0}/chapter'.format(course_directory)
+    os.makedirs(chapter_dir)
+    course_dir = '{0}/course'.format(course_directory)
+    os.makedirs(course_dir)
+    problem_dir = '{0}/problem'.format(course_directory)
+    os.makedirs(problem_dir)
+    seq_dir = '{0}/sequential'.format(course_directory)
+    os.makedirs(seq_dir)
+    vertical_dir = '{0}/vertical'.format(course_directory)
+    os.makedirs(vertical_dir)
+    html_dir = '{0}/html'.format(course_directory)
+    os.makedirs(html_dir)
+
+    course_xml = open('{0}/course.xml'.format(course_dir), "w+")
+    course_xml.write('<course display_name="Course" show_chat="false" enable_timed_exams="true" '
+                     'enable_proctored_exams="false">')
+
+    for chapter in course.assessments:
+        write_chapter(chapter, html_dir, course_xml, chapter_dir, seq_dir, vertical_dir,
+                      problem_dir)
+
+    course_xml.write('</course>')
+    course_xml.close()
+
+
+def parse_assessment_meta(assessment_meta_xml, new_assessment):
+    """
+    Parse metafile for one assessment.
+    """
+    meta_root = assessment_meta_xml.getroot()
+    desc = meta_root.find('meta:description', NS).text
+    if desc is not None:
+        new_assessment.description = re.sub('&lt.*?&gt;|<.*?>|\n', '', desc.encode('utf-8'))
+
+    end_time = meta_root.find('meta:due_at', NS)
+    if end_time is not None:
+        new_assessment.end = '&quot;{0}&quot;'.format(end_time.text)
+    else:
+        end_time = meta_root.find('meta:lock_at', NS)
+        if end_time is not None:
+            new_assessment.end = '&quot;{0}&quot;'.format(end_time.text)

```

Comment:
Why is `&quot;` necessary here?

Reply:
edx fails to upload when date in sequential tag is without quots. interestingly enough in other places edx requires dates without quots